signinapp/admin/users.py

```diff
@@ -149,6 +154,59 @@ def edit_user():
     )
 
 
+@admin.route("/admin/users/edit/student", methods=["GET", "POST"])
+@admin_required
+def edit_student_data():
+    user: User = db.session.get(User, request.args["user_id"])
+    if not user:
+        flash("Invalid user ID")
+        return redirect(url_for("team.users"))
+
+    form = EditStudentDataForm(obj=user)
+
+    if form.validate_on_submit():
+        student_data = user.student_user_data
+        student_data.graduation_year = form.student_data.graduation_year.data
+        student_data.guardians.clear()
```

Comment:
You tested that this doesn't delete the guardian data for some weird reason right?

Reply:
Correct - it just unlinks them, it doesn't delete the guardian